src/hook.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include "common.h"
+#include "posix.h"
+#include "repository.h"
+
+#include "git2/hook.h"
+
+#define GIT_HOOKS_DIR "hooks/"
+
+const char * const githooks[] = {
+	"applypatch-msg",
+	"pre-applypatch",
+	"post-applypatch",
+	"pre-commit",
+	"prepare-commit-msg",
+	"commit-msg",
+	"post-commit",
+	"pre-rebase",
+	"post-checkout",
+	"post-merge",
+	"pre-push",
+	"pre-receive",
+	"update",
+	"post-receive",
+	"post-update",
+	"push-to-checkout",
+	"pre-auto-gc",
+	"post-rewrite",
+};
+
+#define MAX_HOOK_LEN 18+1
+#define HOOK_FILE_MODE 0776
+
+static int build_hook_path(char **hook_path, git_repository *repo, const char *hook_name)
+{
+	int hook_dir_len;
+
+	assert(hook_path);
+
+	hook_dir_len = strlen(repo->path_repository) + strlen(GIT_HOOKS_DIR);
+	*hook_path = git__malloc(hook_dir_len + MAX_HOOK_LEN);
+	GITERR_CHECK_ALLOC(*hook_path);
+
+	strcpy(*hook_path, repo->path_repository);
+	strcat(*hook_path, GIT_HOOKS_DIR);
+	strcat(*hook_path, hook_name);
```

Comment:
Yeah, this thing is 🔥 . I'll try to clean that up, but I still hesitate between `git_buf` and good-ol `char *` for FS.